cloudinit/distros/__init__.py

```diff
@@ -134,9 +151,76 @@ def _unpickle(self, ci_pkl_version: int) -> None:
             # missing expected instance state otherwise.
             self.networking = self.networking_cls()
 
-    @abc.abstractmethod
+    def _extract_package_by_manager(
```

Comment:
These next two methods still feel overly complicated to me. I'm open to any ideas.